cocos/gfx/webgpu/webgpu-commands.ts

```diff
@@ -0,0 +1,1707 @@
+    // dimension optional
+    // let dim: GPUTextureViewDimension = GFXTextureToWebGPUTexture(gpuTexture.type);
+
+    gpuTexture.glTarget = GFXTextureToWebGPUTexture(gpuTexture.type) as GPUTextureViewDimension;
+    gpuTexture.glInternalFmt = GFXFormatToWGPUTextureFormat(gpuTexture.format);
+    gpuTexture.glFormat = GFXFormatToWGPUFormat(gpuTexture.format);
+    gpuTexture.glUsage = GFXTextureUsageToNative(gpuTexture.usage);
+    gpuTexture.glWrapS = gpuTexture.isPowerOf2 ? 'repeat' : 'clamp-to-edge';
+    gpuTexture.glWrapT = gpuTexture.isPowerOf2 ? 'repeat' : 'clamp-to-edge';
+    gpuTexture.glMinFilter = 'linear';
+    gpuTexture.glMagFilter = 'linear';
+    // TBD: 2021 feb 2nd only 1 and 4 supported.
+    gpuTexture.samples = gpuTexture.samples > 1 ? 4 : 1;
+    const texDescriptor: GPUTextureDescriptor = {
+        size: [gpuTexture.width, gpuTexture.height, gpuTexture.arrayLayer],
+        mipLevelCount: gpuTexture.mipLevel,
+        sampleCount: gpuTexture.samples,
+        format: gpuTexture.glFormat,
+        usage: gpuTexture.glUsage,
+    };
+
+    gpuTexture.glTexture = device.nativeDevice!.createTexture(texDescriptor);
+}
+
+export function WebGPUCmdFuncDestroyTexture (gpuTexture: IWebGPUTexture) {
+    if (gpuTexture.glTexture) {
+        gpuTexture.glTexture.destroy();
+    }
+}
+
+export function WebGPUCmdFuncResizeTexture (device: WebGPUDevice, gpuTexture: IWebGPUTexture) {
+    if (gpuTexture.glTexture) {
+        WebGPUCmdFuncDestroyTexture(gpuTexture);
+    }
+    WebGPUCmdFuncCreateTexture(device, gpuTexture);
+}
+
+export function WebGPUCmdFuncCreateSampler (device: WebGPUDevice, gpuSampler: IWebGPUGPUSampler) {
+    const nativeDevice: GPUDevice = device.nativeDevice!;
+
+    gpuSampler.glMinFilter = (gpuSampler.minFilter === Filter.LINEAR || gpuSampler.minFilter === Filter.ANISOTROPIC) ? 'linear' : 'nearest';
+    gpuSampler.glMagFilter = (gpuSampler.magFilter === Filter.LINEAR || gpuSampler.magFilter === Filter.ANISOTROPIC) ? 'linear' : 'nearest';
+    gpuSampler.glMipFilter = (gpuSampler.mipFilter === Filter.LINEAR || gpuSampler.mipFilter === Filter.ANISOTROPIC) ? 'linear' : 'nearest';
+    gpuSampler.glWrapS = WebGPUAdressMode[gpuSampler.addressU];
+    gpuSampler.glWrapT = WebGPUAdressMode[gpuSampler.addressV];
+    gpuSampler.glWrapR = WebGPUAdressMode[gpuSampler.addressW];
+
+    const samplerDesc = {} as GPUSamplerDescriptor;
+    samplerDesc.addressModeU = gpuSampler.glWrapS;
+    samplerDesc.addressModeV = gpuSampler.glWrapT;
+    samplerDesc.addressModeW = gpuSampler.glWrapR;
+    samplerDesc.minFilter = gpuSampler.glMinFilter;
+    samplerDesc.magFilter = gpuSampler.glMagFilter;
+    samplerDesc.mipmapFilter = gpuSampler.glMipFilter;
+    samplerDesc.lodMinClamp = 0;// gpuSampler.minLOD;
+    samplerDesc.lodMaxClamp = gpuSampler.mipLevel;// gpuSampler.maxLOD;
+    // samplerDesc.compare = WebGPUCompareFunc[gpuSampler.compare];
```

Comment:
Is this `compare` assignment needed

Reply:
Adding this breaks all the samplers. I'm not sure if it's a WebGPU issue.

Reply:
If `compare` is defined, it means another type of Sampler: `SamplerComparisonState`, which is different from `SamplerState`.
Currently, if is `gpuSampler.compare` is always, we don't assign `samplerDesc.compare`.